packages/babel-parser/src/plugins/flow.js

```diff
@@ -852,23 +860,37 @@ export default (superClass: Class<Parser>): Class<Parser> =>
             }
           }
 
-          nodeStart.properties.push(
-            this.flowParseObjectTypeProperty(
-              node,
-              isStatic,
-              protoStart,
-              variance,
-              kind,
-              allowSpread,
-            ),
+          const propOrInexact = this.flowParseObjectTypeProperty(
+            node,
+            isStatic,
+            protoStart,
+            variance,
+            kind,
+            allowSpread,
+            allowInexact,
           );
+
+          if (typeof propOrInexact === "boolean") {
+            inexact = inexact || propOrInexact;
+          } else {
+            nodeStart.properties.push(propOrInexact);
+          }
         }
 
         this.flowObjectTypeSemicolon();
       }
 
       this.expect(endDelim);
 
+      /* The inexact flag should only be added on ObjectTypeAnnotations that
+       * are not the body of an interface, declare interface, or declare class.
+       * Since spreads are only allowed in objec types, checking that is
+       * sufficient here.
+       */
+      if (allowSpread) {
```

Comment:
Nit: we should rename this flag, maybe to `isObjectLiteralType`?

Reply:
I agree-- the only reason I'd push back here is because this closely mirrors the way it is implemented in flow, which makes it easier for us to maintain.

https://github.com/facebook/flow/blob/master/src/parser/type_parser.ml#L566-L584

Reply:
Oh ok. It's not a big deal then